test/python/mitigation/test_mitigators.py

```diff
@@ -0,0 +1,293 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# pylint: disable=invalid-name
+
+"""Tests for error mitigation routines."""
+
+import unittest
+from test.python.mitigation.generate_data import test_data
+import numpy as np
+from ddt import ddt, data, unpack
+from qiskit.test import QiskitTestCase
+from qiskit.result import Counts
+from qiskit.mitigation import CorrelatedReadoutMitigator
+from qiskit.mitigation import LocalReadoutMitigator
+from qiskit.mitigation.utils import (
+    z_diagonal,
+    counts_probability_vector,
+    str2diag,
+    expval_with_stddev,
+    stddev,
+)
+from qiskit.test.mock import FakeYorktown
+from qiskit.quantum_info.operators.predicates import matrix_equal
+
+
+@ddt
+class TestReadoutMitigation(QiskitTestCase):
+    """Tests for correlated and local readout mitigation."""
+
+    def compare_results(self, res1, res2):
+        """Compare the results between two runs"""
+        res1_total_shots = sum(res1.values())
+        res2_total_shots = sum(res2.values())
+        keys = set(res1.keys()).union(set(res2.keys()))
+        total = 0
+        for key in keys:
+            val1 = res1.get(key, 0) / res1_total_shots
+            val2 = res2.get(key, 0) / res2_total_shots
+            total += abs(val1 - val2) ** 2
+        return total
+
+    @data([test_data["test_1"]])
```

Comment:
Why not just generate noisy counts directly via multiplying an ideal probability vector by a target A-matrix and sampling noisy counts from that noisy distribution? You dont need a simulator at all.

This means you can directly test mitigation wrt to data generated from the exact A-matrix you use for your mitigator